sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/component.py

```diff
@@ -328,6 +328,7 @@ def _load(
         )
         if yaml_path:
             new_instance._source_path = yaml_path
+            new_instance._base_path = context[BASE_PATH_CONTEXT_KEY]
```

Comment:
why it only applies when yaml_path is provided?

Reply:
uhhh, good catch, I'll move it outer